misc/watcher.py

```diff
@@ -108,9 +111,13 @@ def execute_ocrmypdf(file_path):
         deskew=DESKEW,
         **OCR_JSON_SETTINGS,
     )
-    if exit_code == 0 and ON_SUCCESS_DELETE:
-        log.info(f'OCR is done. Deleting: {file_path}')
-        file_path.unlink()
+    if exit_code == 0:
+        if ON_SUCCESS_DELETE:
+            log.info(f'OCR is done. Deleting: {file_path}')
+            file_path.unlink()
+        elif ON_SUCCESS_ARCHIVE:
+            log.info(f'OCR is done. Archiving {file_path.name} to {ARCHIVE_DIRECTORY}')
+            shutil.move(file_path, f'{ARCHIVE_DIRECTORY}/{file_path.name}')
```

Comment:
I believe we also need to construct any parent directories that don't exist, if /input is nested.

Reply:
Also, perhaps the archive directory should have the same year-month structure as the output directory structure if that flag is set.

Reply:
> I believe we also need to construct any parent directories that don't exist, if /input is nested.

Do you mean that `<input>/in/nested/directories/input.pdf` should be archived to `<archive>/in/nested/directories/input.pdf`?

And if OUTPUT_DIRECTORY_YEAR_MONTH is set, would it be either `<archive>/<year>/<month>/in/nested/directories/input.pdf` or `<archive>/in/nested/directories/<year>/<month>/input.pdf`?

For me, both option seem confusing, so I'd probably strip the nested directories and simply go with `<archive>/<year>/<month>/input.pdf`, risking to overwrite existing files.

Reply:
> Do you mean that <input>/in/nested/directories/input.pdf should be archived to <archive>/in/nested/directories/input.pdf?

And if so, shouldn't the same be true for the output path, for consistency? Currently, nested directory paths are stripped from the output path as well.

Reply:
Okay... never mind nesting directory paths then if we're not doing that already.